WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderCommentListViewModel.kt

```diff
@@ -31,4 +65,77 @@ class ReaderCommentListViewModel
     }
 
     data class ScrollPosition(val position: Int, val isSmooth: Boolean)
+
+    fun start(blogId: Long, postId: Long) {
+        if (isStarted) return
+        isStarted = true
+
+        this.blogId = blogId
+        this.postId = postId
+
+        _updateFollowStatus.value = FollowCommentsNotAllowed
+
+        init()
+    }
+
+    private fun init() {
+        _snackbarEvents.addSource(followCommentsHandler.snackbarEvents) { event ->
+            _snackbarEvents.value = event
+        }
+
+        _updateFollowStatus.addSource(followCommentsHandler.followStatusUpdate) { event ->
+            _updateFollowStatus.value = event
+        }
+
+        getFollowConversationStatus(blogId, postId, true)
+    }
+
+    private fun onFollowConversationClicked(askSubscribe: Boolean) {
+        if (!followUnfollowCommentsFeatureConfig.isEnabled()) return
+        followStatusSetJob?.cancel()
+        followStatusSetJob = launch(bgDispatcher) {
+            followCommentsHandler.handleFollowCommentsClicked(blogId, postId, askSubscribe)
+        }
+    }
+
+    private fun getFollowConversationStatus(blogId: Long, postId: Long, isInit: Boolean) {
+        if (!followUnfollowCommentsFeatureConfig.isEnabled()) return
+        followStatusGetJob?.cancel()
+        followStatusGetJob = launch(bgDispatcher) {
+            followCommentsHandler.handleFollowCommentsStatusRequest(blogId, postId, isInit)
+        }
+    }
+
+    private fun buildFollowCommentsUiState(followCommentsState: FollowCommentsState): FollowCommentsUiState {
+        return if ((followCommentsState is NoNetwork && !followCommentsState.isGetStatus) ||
+                (followCommentsState is Failure && !followCommentsState.isGetStatus)) {
+            Nop
```

Comment:
I think I'd personally move this check to the place where we set the value of the live data. For example we could filter it out here: 
```
_updateFollowStatus.addSource(followCommentsHandler.followStatusUpdate) { event ->
            _updateFollowStatus.value = event
        }
```

It feels a little out of place to set the field and not use the value in certain cases.

Reply:
Agree. Removed the Nop completely here 015dd2a8c791d38eb3427bb80fbcb04d8ef3d329